packages/mobile/src/escrow/saga.ts

```diff
@@ -268,16 +267,20 @@ function* doFetchSentPayments() {
     const sentPaymentsRaw = yield all(
       sentPaymentIDs.map((paymentID) => call(getEscrowedPayment, escrow, paymentID))
     )
-    const tempAddresstoRecipientPhoneNumber: Invitees = yield select(inviteesSelector)
+
+    const addressToE164Number = yield select(addressToE164NumberSelector)
     const sentPayments: EscrowedPayment[] = []
     for (let i = 0; i < sentPaymentsRaw.length; i++) {
-      const id = sentPaymentIDs[i].toLowerCase()
-      const recipientPhoneNumber = tempAddresstoRecipientPhoneNumber[id]
+      const address = sentPaymentIDs[i]
+      const recipientPhoneNumber = addressToE164Number[address]
+      const id = sentPaymentIDs[i].toLowerCase() // OPEN QUESTION: why are we making this lowercase? seems inconsistent
       const payment = sentPaymentsRaw[i]
       if (!payment) {
         continue
       }
 
+      // OPEN QUESTION: EscrowedPayment and InviteDetails appear to store redundant data but invitee data is stored on send and escrow data is fetched from the smart contract.
```

Comment:
I don't think we can fetch the phone number from the contract, right?

Reply:
Another option to eliminate redundancy between EscrowedPayment and InviteDetails is to eliminate tx related data (timestamp, escrowAmount, escrowTxId, etc) from InviteDetails and just do a join on the two structures when we need to. What do you think? Is there ever a time when InviteDetails would have tx data we couldn't get from the smart contract?